File: src/main/java/macro303/keschet/core/players/Reader.java
package macro303.keschet.core.players;

import macro303.board_game.Colour;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.jetbrains.annotations.NotNull;

import java.util.Scanner;

/**
 * Created by Macro303 on 2018-02-08.
 */
public abstract class Reader {
	private static final Logger LOGGER = LogManager.getLogger(Reader.class);
	@NotNull
	private static final Scanner reader = new Scanner(System.in);

	@NotNull
	public static String readConsole(String prompt) {
		System.out.print(Colour.WHITE.getColourCode());
		System.out.println(prompt);
		System.out.print("> ");
		System.out.print(Colour.RESET.getColourCode());
		return reader.nextLine().trim();
	}
}

File: Base/src/main/kotlin/macro303/base/Reader.kt
package macro303.base

import org.apache.logging.log4j.LogManager
import java.util.*

class Reader {
	private val reader = Scanner(System.`in`)

	fun readConsole(prompt: String): String {
		Thread.sleep(10)
		println(prompt)
		print("> ")
		val input = reader.nextLine().trim()
		LOGGER.trace("String readConsole(String) = $input")
		return input
	}

	companion object {
		private val LOGGER = LogManager.getLogger(Reader::class.java)
	}
}
